pkg/client/unversioned/helper.go

```diff
@@ -290,7 +291,8 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if len(config.Version) == 0 {
+	if len(config.Version) == 0 || apiutil.GetGroup(config.Version) != "" {
```

Comment:
on second thought, this won't completely fix the problem. If you take a look at https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/helper.go#L136, the RESTClient and the ExperimentalClient share the same config, so if you set the config.Version here, the ExperimentalClient will not have the config.